src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -54,13 +56,34 @@ public static Command parse(String input) throws DukeException {
         case COMMAND_LOGIN:
             checkLogInValidity(parsedInputs);
             return new LogInCommand(parsedInputs[1]);
+        case COMMAND_ADD:
+            checkAddValidity(parsedInputs);
+            return new AddCommand(parsedInputs[1]);
         case COMMAND_BYE:
             return new ByeCommand();
         default:
             throw new DukeException("Sorry! I don't know what that means :-(");
         }
     }
 
+    private static void checkAddValidity(String[] input) throws DukeException {
+        if (input.length < 2) {
+            throw new DukeException("There is no description in your add command!");
+        } else if (!input[1].contains("/")) {
+            throw new DukeException("An add command needs to be in a 'name /day /time /location' format!");
+        }
+        String[] position = input[1].split(" /",4);
+        if (position[0].isEmpty()) {
+            throw new DukeException("There is no name in your add command!");
+        } else if (position[1].isEmpty()) {
+            throw new DukeException("There is no day in your add command!");
+        } else if (position[2].isEmpty()) {
+            throw new DukeException("There is no time in your add command!");
+        } else if (position[3].isEmpty()) {
+            throw new DukeException("There is no location in your add command!");
+        }
+
+    }
```

Comment:
Please consider using switch-case. Otherwise, the last one should be a single else-statement. (https://se-education.org/guides/conventions/java/basic.html#layout)

Reply:
Hi, I have made the changes mentioned above and updated it. Kindly check if you can merge it. thank you!